### ISSUE TITLE
provide a way to figure out an existing grafana folder_id

### ISSUE THREAD

**Issue by rd-michel** [grafana/terraform-provider-grafana#219]
Suggesting a new feature to support querying the existing grafana instance to determine an existing grafana folder_id (native data source).

the current "terraform-provider-grafana" does not provide any data source. we can only create new resources (afaik). is that correct?

it should be possible by using the current grafana api (but im not a developer)
https://grafana.com/docs/grafana/latest/http_api/folder/
https://grafana.com/docs/grafana/latest/http_api/folder_dashboard_search/

example code (imagine two seperate deployments)

**deployment 1:**

tasks:
- create grafana_folder
- create a new grafana dashboard within that folder

```
resource "grafana_folder" "collection" {
  title = "team-example"
}

resource "grafana_dashboard" "metrics" {
  folder      = grafana_folder.collection.id
  config_json = file("grafana-dashboard.json")
}
```

**deployment 2:** 

tasks:
- find the folder_id of an existing folder
- create a grafana dashboard within that folder

```
data "grafana_folder" "collection" {
  title = "team-example"
}

resource "grafana_dashboard" "metrics" {
  folder      = data.grafana_folder.collection.id
  config_json = file("grafana-dashboard.json")
}
```

**Comment by trotttrotttrott** [comment]
Would love to have corresponding data sources for all resources. It should be fairly strait-forward to implement. For the time being you can import folders that were created outside of Terraform.

For example, if you already have a folder called, "team-example", you can create a resource config for it:

```hcl
resource "grafana_folder" "collection" {
  title = "team-example"
}
```

...and then import it with `terraform import grafana_folder.collection {{folder_id}}`.

https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/folder#import

**Comment by rd-michel** [comment]
we know that there are a bunch of workarounds 

- use remote state
- use terraform-provider-http to access the grafana-api
- manually import the folder

...but in the  end these are all workarounds due to the lack of a "data source". the manual import is one of the worst "solutions" cause it breaks the whole idea of automating these steps.

**Comment by stevenproctor** [comment]
Would you have a workaround in the mean time for folder datasource without importing the resource?

For clarity: we have multiple dashboards in different repos for different services that live under a same folder, so importing would have multiple Terraform modules try and take ownership of that folder.

In the mean time we are doing a folder for every dashboard, but that feels like overkill because of the lack of a datasource.

**Comment by justinTM** [comment]
implemented via #312
